src/deepLink/deepLinkService.ts

```diff
@@ -0,0 +1,293 @@
+import type { Disposable } from 'vscode';
+import { window } from 'vscode';
+import { Commands } from '../constants';
+import type { Container } from '../container';
+import { GitReference } from '../git/models/reference';
+import type { GitRemote } from '../git/models/remote';
+import type { ShowInCommitGraphCommandArgs } from '../plus/webviews/graph/graphWebview';
+import { executeCommand } from '../system/command';
+import type { UriEvent } from '../uri/uri';
+import { UriTypes } from '../uri/uri';
+import type { DeepLinkServiceContext } from './deepLink';
+import { DeepLinkServiceAction, DeepLinkServiceState, deepLinkStateTransitionTable, DeepLinkType } from './deepLink';
+
+export class DeepLinkService implements Disposable {
+	private _disposable: Disposable;
+	private _context: DeepLinkServiceContext;
+
+	constructor(private readonly container: Container) {
+		this._context = {
+			state: DeepLinkServiceState.Idle,
+		};
+
+		this._disposable = container.uri.onDidReceiveUri(async (event: UriEvent) => {
+			if (event.type === UriTypes.DeepLink && this._context.state === DeepLinkServiceState.Idle) {
+				if (!event.repoId || !event.linkType || !event.uri || !event.remoteUrl) {
+					void window.showErrorMessage(`Error resolving deep link: missing required properties.`);
+					return;
+				}
+
+				if (!Object.values(DeepLinkType).includes(event.linkType)) {
+					void window.showErrorMessage(`Error resolving deep link: unknown link type.`);
+					return;
+				}
+
+				if (event.linkType !== DeepLinkType.Remote && !event.targetId) {
+					void window.showErrorMessage(
+						`Error resolving deep link of type ${event.linkType}: no target id provided.`,
```

Comment:
Maybe just "Unable to resolve link"? I know it helps for debugging (and would output it to the log with more details, but unsure if it helps the user)

Reply:
Done in commit: [Log deep link issues](https://github.com/gitkraken/vscode-gitlens/pull/2471/commits/8e25b9de822b6c0be4ff9fb958d12d02d0ea494b)